src/esp/assets/ResourceManager.cpp

```diff
@@ -1648,43 +1751,257 @@ void ResourceManager::loadMaterials(Importer& importer,
+                 BCTexture);
+      if (metalness >= 0.5) {
+        safeAppend(origMaterialData, newAttributes, MnMatAttr::SpecularTexture,
+                   BCTexture);
+      }
+    }
+  }  // if no phong material support exists in material
+
+  if (~(origMatTypes & Mn::Trade::MaterialType::PbrMetallicRoughness)) {
+    // add appropriate values for expected attributes for PbrMetallicRoughness
+    // derived from Phong attributes
+    const auto& phongMaterial =
+        origMaterialData->as<Mn::Trade::PhongMaterialData>();
+
+    // derive base color from Phong diffuse or ambient color, depending on which
+    // is present.  set to white if neither is present
+    const Mn::Color4 baseColor = phongMaterial.diffuseColor();
+
+    // Experimental metalness heuristic using saturation of spec color
+    // to derive approximation of metalness
+    const Mn::Color4 specColor = phongMaterial.specularColor();
+
+    // if specColor alpha == 0 then no metalness
+    float metalness = 0.0f;
+    // otherwise, this heuristic will derive a value for metalness based on
+    // how non-grayscale the specular color is (HSV Saturation).
+    if (specColor.a() != 0.0f) {
+      metalness = specColor.toHsv().saturation;
+    }
+    /////////////////
+    // set PbrMetallicRoughness attributes appropriately from precalculated
+    // values
+    ////////////////
+    // normal mapping is already present in copied array if present in
+    // original material.
+    safeAppend(origMaterialData, newAttributes, MnMatAttr::BaseColor,
+               baseColor);
+    safeAppend(origMaterialData, newAttributes, MnMatAttr::Metalness,
+               metalness);
+
+    // texture transforms, if there's none the returned matrix is an
+    // identity Only copy if we don't already have TextureMatrix attribute
+    // (if original phongMaterial does not have that specific attribute)
+
+    ESP_WARNING() << "before Phong texture matrix addition";
+    if (!setTexMatrix &&
+        (!phongMaterial.hasAttribute(MnMatAttr::TextureMatrix))) {
+      arrayAppend(newAttributes, {MnMatAttr::TextureMatrix,
+                                  phongMaterial.commonTextureMatrix()});
+      setTexMatrix = true;
+    }
+    ESP_WARNING() << "after Phong texture matrix addition";
+    // base texture
+
+  }  // if no PbrMetallicRoughness material support exists in material
+
+  // build flags to support all materials
+  const auto flags = Mn::Trade::MaterialType::Flat |
+                     Mn::Trade::MaterialType::Phong |
+                     Mn::Trade::MaterialType::PbrMetallicRoughness;
+
+  // create new material from attributes array
+  ESP_WARNING() << "before building new material";
+  Mn::Trade::MaterialData newMaterialData{flags, std::move(newAttributes)};
+  ESP_WARNING() << "after building new material";
+  // move into optional container and return
+  return Cr::Containers::Optional<Mn::Trade::MaterialData>{
```

Comment:
Same here, why wrap the return value in an optional when it's never going to be null? Designing the API this way implies that the caller should except the operation to fail and check if the return value is valid, but that never actually happens.